src/js/components/time-picker.js

```diff
@@ -0,0 +1,134 @@
+const behavior = require("../utils/behavior");
+const select = require("../utils/select");
+const { prefix: PREFIX } = require("../config");
+const { COMBO_BOX_CLASS, enhanceComboBox } = require("./combo-box");
+
+const TIME_PICKER_CLASS = `${PREFIX}-time-picker`;
+const TIME_PICKER = `.${TIME_PICKER_CLASS}`;
+const MAX_TIME = 60 * 24 - 1;
+const MIN_TIME = 0;
+const DEFAULT_STEP = 30;
+const MIN_STEP = 1;
+
+const FILTER_DATASET = {
+  filter:
+    "0?{{ hourQueryFilter }}:{{minuteQueryFilter}}.*{{ apQueryFilter }}m?",
+  apQueryFilter: "([ap])",
+  hourQueryFilter: "([1-9][0-2]?)",
```

Comment:
I confess I don't know if this messes with your capture groups, but:
```
  const VALID_HOUR_RE = /^((0?[1-9])|([1][0-2]))$/; // 1-9, 01-09, 10-12
  const VALID_MINUTE_RE = /^([0-5][0-9])$/; // 00-59
```

Reply:
At the end of the day these are not validating regular expressions; instead these are query filters to help build a dynamic regular expression that will then try to match against a combo box option.

So even though the regular expression query filter captures a lot it wouldn't match against anything in reality. 

Also, I believe that the one digit minute is needed for (04:1p) matching to (04:15pm)

Reply:
That makes sense.  Thanks for the insight!